src/chainstate/burn/operations/peg_out_request.rs

```diff
@@ -0,0 +1,384 @@
+        */
+
+        if data.len() < 73 {
+            // too short
+            warn!(
+                "PegOutRequestOp payload is malformed ({} bytes, expected {})",
+                data.len(),
+                73
+            );
+            return Err(ParseError::MalformedPayload);
+        }
+
+        let amount = u64::from_be_bytes(data[0..8].try_into().unwrap());
+        let signature = MessageSignature::from_bytes(&data[8..73]).unwrap();
+        let memo = data.get(73..).unwrap_or(&[]).to_vec();
+
+        Ok(ParsedData {
+            amount,
+            signature,
+            memo,
+        })
+    }
+
+    pub fn check(&self) -> Result<(), OpError> {
+        if self.amount == 0 {
+            warn!("PEG_OUT_REQUEST Invalid: Requested BTC amount must be positive");
+            return Err(OpError::AmountMustBePositive);
+        }
+
+        if self.fulfillment_fee == 0 {
+            warn!("PEG_OUT_REQUEST Invalid: Fulfillment fee must be positive");
+            return Err(OpError::AmountMustBePositive);
+        }
+
+        Ok(())
+    }
+}
+
+struct ParsedData {
+    amount: u64,
+    signature: MessageSignature,
+    memo: Vec<u8>,
+}
+
+enum ParseError {
+    MalformedPayload,
+    SliceConversion,
+}
+
+impl From<ParseError> for OpError {
+    fn from(_: ParseError) -> Self {
+        Self::ParseError
+    }
+}
+
+impl From<std::array::TryFromSliceError> for ParseError {
+    fn from(_: std::array::TryFromSliceError) -> Self {
+        Self::SliceConversion
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::chainstate::burn::operations::test;
+
+    use super::*;
+
+    #[test]
+    fn test_parse_peg_out_request_should_succeed_given_a_conforming_transaction() {
+        let mut rng = test::seeded_rng();
+        let opcode = Opcodes::PegOutRequest;
+
+        let dust_amount = 1;
+        let recipient_address_bytes = test::random_bytes(&mut rng);
+        let output2 = test::Output::new(dust_amount, recipient_address_bytes);
+
+        let peg_wallet_address = test::random_bytes(&mut rng);
+        let fulfillment_fee = 3;
+        let output3 = test::Output::new(fulfillment_fee, peg_wallet_address);
+
+        let mut data = vec![];
+        let amount: u64 = 10;
+        let signature: [u8; 65] = test::random_bytes(&mut rng);
+        data.extend_from_slice(&amount.to_be_bytes());
+        data.extend_from_slice(&signature);
+
+        let tx = test::burnchain_transaction(data, [output2, output3], opcode);
+        let header = test::burnchain_block_header();
+
+        let op =
+            PegOutRequestOp::from_tx(&header, &tx).expect("Failed to construct peg-out operation");
+
+        assert_eq!(op.recipient.bytes(), recipient_address_bytes);
+        assert_eq!(op.signature.as_bytes(), &signature);
+        assert_eq!(op.amount, amount);
+    }
+
+    #[test]
+    fn test_parse_peg_out_request_should_succeed_given_a_conforming_transaction_with_extra_memo_bytes(
```

Comment:
Minor formatting nit -- please remove the newline.  `()` belongs on one line.

Reply:
Done :heavy_check_mark: